PlayCover/AppInstaller/Binary Patcher/Operations.swift

```diff
@@ -0,0 +1,208 @@
+        commandData.append(dylibPath.data(using: .ascii)!)
+        commandData.append(&zeroByte, length: Int(padding))
+
+        binary.replaceBytes(in: NSRange(location: Int(macho.offset + macho.header.sizeofcmds + macho.size),
+                                        length: commandData.length), withBytes: nil, length: 0)
+
+        binary.replaceBytes(in: NSRange(location: Int(lastOffset),
+                                        length: 0), withBytes: commandData.bytes, length: commandData.length)
+
+        macho.header.ncmds += 1
+        macho.header.sizeofcmds += command.cmdsize
+
+        binary.replaceBytes(in: NSRange(location: Int(macho.offset),
+                                        length: MemoryLayout.size(ofValue: macho.header)), withBytes: &macho.header)
+
+        return true
+    }
+
+    @discardableResult
+    public static func removeLoadEntryFromBinary(_ binary: inout NSMutableData,
+                                                 _ macho: inout thin_header,
+                                                 _ payload: String) -> Bool {
+        var offset: Int = Int(macho.offset + macho.size)
+
+        var num: UInt32 = 0
+        var cumulativeSize: UInt32 = 0
+        var removedOrdinal: Int32 = -1
+
+        for index in 0..<macho.header.ncmds {
+            if offset >= binary.length || offset > macho.offset + macho.size + macho.header.sizeofcmds {
+                break
+            }
+
+            let cmdData = Data(bytes: binary[offset..<offset+4].base.bytes, count: 4)
+            let cmd = cmdData.withUnsafeBytes { $0.load(as: UInt32.self) }
+
+            let sizeData = Data(bytes: binary[offset+4..<offset+8].base
+                .bytes, count: 4)
+            let size = sizeData.withUnsafeBytes { $0.load(as: UInt32.self) }
+
+            switch cmd {
+            case LC_REEXPORT_DYLIB,
+            LC_LOAD_UPWARD_DYLIB,
+            LC_LOAD_WEAK_DYLIB,
+            UInt32(LC_LOAD_DYLIB):
+                let command: dylib_command = (binary.bytes + offset).load(as: dylib_command.self)
+                let bytes = binary.subdata(with:
+                                            NSRange(location: offset + Int(command.dylib.name.offset),
+                                                    length: Int(command.cmdsize - command.dylib.name.offset)))
+                    .withUnsafeBytes({ $0.load(as: Array<UInt8>.self) })
+                let name = String(bytes: bytes, encoding: .utf8)
+                if name == payload && removedOrdinal == -1 {
+                    print("Removing payload from \(LC(cmd))...")
+                    binary.replaceBytes(in: NSRange(location: offset,
+                                                    length: Int(size)), withBytes: nil, length: 0)
+                    num += 1
+                    cumulativeSize += size
+                    removedOrdinal = Int32(index)
+                }
+
+                offset += Int(size)
+            default:
```

Comment:
have you missed a case branch?

Reply:
That case is only run in debug

Reply:
The `#ifdef DEBUG` macro only wraps a part of that case. The remaining part still runs even if not in debug.